### PR TITLE
Add Brave browser plugin

### PR DESCRIPTION
**Pull request by JSCU-CNI**
This PR adds a `BrowserPlugin` for Brave. 

While working on this PR we found that cookies are stored in the `Network` subfolder in newer versions of Chromium-based browsers.  We therefore made a small addition to `_iter_db` to support checking additional subfolders, which can then be used in the `cookies` method.

### CODE REVIEW

File: dissect/target/plugins/apps/browser/chromium.py [hunk 1]
```diff
@@ -69,11 +70,12 @@ def _build_userdirs(self, hist_paths: list[str]) -> list[tuple[UserDetails, Targ
                 users_dirs.append((user_details.user, cur_dir))
         return users_dirs
 
-    def _iter_db(self, filename: str) -> Iterator[SQLite3]:
+    def _iter_db(self, filename: str, *subfolders: Optional[list[str]]) -> Iterator[SQLite3]:
```
Comment (Schamper) [inline dissect/target/plugins/apps/browser/chromium.py:73]:
```suggestion
    def _iter_db(self, filename: str, subdirs: Optional[list[str]] = None) -> Iterator[SQLite3]:
```
I don't think using `*` is the nicest approach here. I feel like an explicit list of sub directories is better suited.

I've also renamed it to use the `dir` name, mostly for consistency with the other methods in this file.

File: dissect/target/plugins/apps/browser/chromium.py [hunk 2]
```diff
@@ -69,11 +70,12 @@ def _build_userdirs(self, hist_paths: list[str]) -> list[tuple[UserDetails, Targ
...
         """Generate a connection to a sqlite database file.
 
         Args:
             filename: The filename as string of the database where the data is stored.
+            subfolders (optional): Subfolder(s) to also try for every configured directory.
```
Comment (Schamper) [inline dissect/target/plugins/apps/browser/chromium.py:78]:
```suggestion
            subdirs: Subdirectories to also try for every configured directory.
```

File: dissect/target/plugins/apps/browser/chromium.py [hunk 3]
```diff
@@ -83,7 +85,11 @@ def _iter_db(self, filename: str) -> Iterator[SQLite3]:
             SQLError: If the history file could not be opened.
         """
 
-        for user, cur_dir in self._build_userdirs(self.DIRS):
+        dirs = self.DIRS
+        if subfolders:
+            dirs = dirs + [join(dir, subdir) for dir, subdir in itertools.product(self.DIRS, subfolders)]
```
Comment (Schamper) [inline dissect/target/plugins/apps/browser/chromium.py:90]:
```suggestion
        if subdirs:
            dirs.extend([join(dir, subdir) for dir, subdir in itertools.product(self.DIRS, subdirs)])
```

File: dissect/target/plugins/apps/browser/chromium.py [hunk 4]
```diff
@@ -198,7 +204,7 @@ def cookies(self, browser_name: Optional[str] = None) -> Iterator[BrowserCookieR
                 is_http_only (bool): Cookie http only flag.
                 same_site (bool): Cookie same site flag.
         """
-        for user, db_file, db in self._iter_db("Cookies"):
+        for user, db_file, db in self._iter_db("Cookies", "Network"):
```
Comment (Schamper) [inline dissect/target/plugins/apps/browser/chromium.py:207]:
```suggestion
        for user, db_file, db in self._iter_db("Cookies", subdirs=["Network"]):
```

File: dissect/target/plugins/apps/browser/brave.py [hunk 5]
```diff
@@ -0,0 +1,69 @@
...
+
+
+class BravePlugin(ChromiumMixin, BrowserPlugin):
+    """Brave browser plugin."""
+
+    __namespace__ = "brave"
+
+    DIRS = [
+        # Windows
+        "AppData/Local/BraveSoftware/Brave-Browser/User Data/Default",
```
Comment (Schamper) [inline dissect/target/plugins/apps/browser/brave.py:26]: Could you please also add these paths to acquire? :smile:

**Review by Schamper — CHANGES_REQUESTED** [review]

File: dissect/target/plugins/apps/browser/brave.py (hunk 5 above)
Comment (JSCU-CNI) [reply dissect/target/plugins/apps/browser/brave.py:26]: Added in https://github.com/fox-it/acquire/pull/129.

File: dissect/target/plugins/apps/browser/brave.py [hunk 6]
```diff
@@ -0,0 +1,69 @@
...
+        "Library/Application Support/BraveSoftware/Default",
+    ]
+
+    BrowserHistoryRecord = create_extended_descriptor([UserRecordDescriptorExtension])(
+        "browser/brave/history", GENERIC_HISTORY_RECORD_FIELDS
+    )
+
+    BrowserCookieRecord = create_extended_descriptor([UserRecordDescriptorExtension])(
+        "browser/brave/cookie",
+        GENERIC_COOKIE_FIELDS,
```
Comment (Schamper) [inline dissect/target/plugins/apps/browser/brave.py:40]:
```suggestion
        "browser/brave/cookie", GENERIC_COOKIE_FIELDS
```

**Review by Schamper — CHANGES_REQUESTED** [review]

**Review by Schamper — APPROVED** [review]

**Pull request merged**